x/evm/types/compiled_contract.go

```diff
@@ -69,27 +70,57 @@ func (s *CompiledContract) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// HardhatCompiledContract is a type used to unpack the compiled JSON data
+// which is generated by running `npx hardhat compile` for a given smart contract.
+type HardhatCompiledContract struct {
+	Format       string  `json:"_format"`
+	ContractName string  `json:"contractName"`
+	SourceName   string  `json:"sourceName"`
+	ABI          abi.ABI `json:"abi"`
+	Bytecode     string  `json:"bytecode"`
+}
+
+func (c HardhatCompiledContract) ToCompiledContract() (CompiledContract, error) {
+	strippedHex := strings.TrimPrefix(c.Bytecode, "0x")
+	hexBytes, err := hex.DecodeString(strippedHex)
+	if err != nil {
+		return CompiledContract{}, fmt.Errorf("failed to decode hex string: %w", err)
+	}
+
+	return CompiledContract{
+		ABI: c.ABI,
+		Bin: hexBytes,
+	}, nil
+}
+
 var (
-	//go:embed ERC20Contract.json
+	//go:embed testdata/ERC20Contract.json
 	erc20JSON []byte
 
 	// ERC20Contract is the compiled test erc20 contract
 	ERC20Contract CompiledContract
 
-	//go:embed SimpleStorageContract.json
+	//go:embed testdata/SimpleContractHardhat.json
+	SimpleContractJSON []byte
+
+	// SimpleContract is the compiled test simple contract
+	SimpleContract HardhatCompiledContract
+
+	//go:embed testdata/SimpleStorageContract.json
 	simpleStorageJSON []byte
 
 	// SimpleStorageContract is the compiled test simple storage contract
 	SimpleStorageContract CompiledContract
 
-	//go:embed TestMessageCall.json
+	//go:embed testdata/TestMessageCall.json
 	testMessageCallJSON []byte
 
 	// TestMessageCall is the compiled message call benchmark contract
 	TestMessageCall CompiledContract
 )
 
 func init() {
+	// Unmarshal the compiled contracts in the legacy format
```

Comment:
It makes some noise to me that in this file we're mixing functions and variables that are only used on tests and some on the protocol. For example, this init function is invoked on `evmosd start` command which does not make sense.

Probably would be better to move all the testing stuff to the `testutil` directory and leave only the required types, vars and funcs used in the protocol

Reply:
I agree, good point! 👍